internal/pkg/cli/pipeline_update.go

```diff
@@ -0,0 +1,245 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package cli
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/archer"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/aws/identity"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/aws/session"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/deploy"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/deploy/cloudformation"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/manifest"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/store"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/term/color"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/term/log"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/workspace"
+
+	"github.com/aws/aws-sdk-go/aws"
+
+	termprogress "github.com/aws/amazon-ecs-cli-v2/internal/pkg/term/progress"
+	"github.com/spf13/cobra"
+)
+
+const (
+	fmtAddPipelineResourcesStart    = "Adding pipeline resources to your project: %s"
+	fmtAddPipelineResourcesComplete = "Successfully added pipeline resources to your project: %s"
+
+	fmtUpdatePipelineFailed   = "Failed to accept changes for pipeline: %s."
+	fmtUpdatePipelineStart    = "Proposing infrastructure changes for the pipeline: %s"
+	fmtUpdatePipelineComplete = "Successfully updated pipeline: %s"
+)
+
+var errNoPipelineFile = errors.New("There was no pipeline manifest found in your workspace. Please run `archer pipeline init` to create an pipeline.")
+
+// UpdatePipelineOpts holds the configuration needed to create or update a pipeline
+type UpdatePipelineOpts struct {
+	// Fields with matching flags.
+	PipelineFile string
+	// Deploy bool
+
+	pipelineDeployer pipelineDeployer
+	project          *archer.Project
+	prog             progress
+	region           string
+	envStore         archer.EnvironmentStore
+	account          string
+	ws               archer.Workspace
+
+	*GlobalOpts
+}
+
+func NewUpdatePipelineOpts() *UpdatePipelineOpts {
+	return &UpdatePipelineOpts{
+		GlobalOpts: NewGlobalOpts(),
+		prog:       termprogress.NewSpinner(),
+	}
+}
+
+// func (opts *UpdatePipelineOpts) Ask() error {
+// 	if opts.PipelineFile() == "" {
+// 		return errNoPipelineFile
+// 	}
+
+// 	return nil
+// }
+
+// Validate returns an error if the flag values passed by the user are invalid.
+func (opts *UpdatePipelineOpts) Validate() error {
+	if opts.PipelineFile == "" {
+		return errNoPipelineFile
+	}
+
+	return nil
+}
+
+func (opts *UpdatePipelineOpts) convertStages(manifestStages []manifest.PipelineStage) ([]deploy.PipelineStage, error) {
+	stages := []deploy.PipelineStage{}
```

Comment:
nit: the recommended way of declaring slices is: `var stages []deploy.PipelineStage` (https://github.com/golang/go/wiki/CodeReviewComments#declaring-empty-slices)

Reply:
I TOLD YOU @SoManyHs !!!!